source/server/connection_handler_impl.h

```diff
@@ -47,37 +49,43 @@ class ConnectionHandlerImpl : public Network::ConnectionHandler, NonCopyable {
 
   // Network::ConnectionHandler
   uint64_t numConnections() override { return num_connections_; }
-  void addListener(Network::FilterChainFactory& factory, Network::ListenSocket& socket,
-                   Stats::Scope& scope, uint64_t listener_tag,
-                   const Network::ListenerOptions& listener_options) override;
-  void addSslListener(Network::FilterChainFactory& factory, Ssl::ServerContext& ssl_ctx,
-                      Network::ListenSocket& socket, Stats::Scope& scope, uint64_t listener_tag,
-                      const Network::ListenerOptions& listener_options) override;
-  Network::Listener* findListenerByAddress(const Network::Address::Instance& address) override;
+  void addListener(Listener& config) override;
   void removeListeners(uint64_t listener_tag) override;
   void stopListeners(uint64_t listener_tag) override;
   void stopListeners() override;
 
+  Network::Listener* findListenerByAddress(const Network::Address::Instance& address) override;
+
 private:
+  struct ActiveListener;
+  ActiveListener* findActiveListenerByAddress(const Network::Address::Instance& address);
+
   struct ActiveConnection;
   typedef std::unique_ptr<ActiveConnection> ActiveConnectionPtr;
+  struct ActiveSocket;
+  typedef std::unique_ptr<ActiveSocket> ActiveSocketPtr;
 
   /**
    * Wrapper for an active listener owned by this handler.
    */
   struct ActiveListener : public Network::ListenerCallbacks {
-    ActiveListener(ConnectionHandlerImpl& parent, Network::ListenSocket& socket,
-                   Network::FilterChainFactory& factory, Stats::Scope& scope, uint64_t listener_tag,
-                   const Network::ListenerOptions& listener_options);
+    ActiveListener(ConnectionHandlerImpl& parent, Listener& config);
 
     ActiveListener(ConnectionHandlerImpl& parent, Network::ListenerPtr&& listener,
-                   Network::FilterChainFactory& factory, Stats::Scope& scope,
-                   uint64_t listener_tag);
+                   Listener& config);
 
     ~ActiveListener();
 
+    // Network::ListenerCallbacks
     /**
-     * Fires when a new connection is received from the listener.
+     * Fires when a new accepted socket is received from the listener.
```

Comment:
nit: delete doc comments for overridden functions

Reply:
OK.